src/markdown/generator.ml

```diff
@@ -0,0 +1,310 @@
+open Odoc_document
+open Types
+open Doctree
+
+module Markup = struct
```

Comment:
What about making `t` abstract ? Writing down the types of all the functions signatures would make it more readable. Also, it would allow internal functions to the `Markup` module that are inaccessible outside of it.
You'll need a function `let string s = String s` (or an other name) because there's some uses of `String` later. (same for `Anchor`).

Reply:
> What about making t abstract ?
 

I’m not sure about your suggestion!

Reply:
Making the type `t` abstract will help split the concern between `Markup` and the rest of the module and also improve readability. Like this:

```ocaml
module Markup : sig

  type t

  (* The signature of other functions *)

end = struct
```

That way it's easier to understand the rest of the code by looking only at that interface but also it's easier to modify and document `Markup`.